### ISSUE TITLE
NVDA doesn't announce text defined in next div after aria-hidden element.

### ISSUE THREAD

**Issue by msftedad** [nvaccess/nvda#13757]
### Steps to reproduce:
1. Start NVDA. Open codepen https://codepen.io/testaccount01/pen/jOZzXpG
2. Turn off browse mode (Using Insert/caps Lock + Space bar)
3. Now navigate using up\down arrow key and observe the issue.

### Actual behavior:
Screen reader doesn't announce the date in single keystroke when aria-hidden="true" is defined and also the last line placed after date is not accessible when browse mode is off.

### Expected behavior:
Screen reader should announce the date in single keystroke and it should also announce the last line placed after date.

### System configuration
Windows 11

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA Version 2022.1

#### Windows version:
OS build: 21H2 (OS Build 22000.675)

#### Name and version of other software in use when reproducing the issue:
Edge Version 101.0.1210.53

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, NVDA Version 2021.3.5

**Comment by feerrenrut** [comment]
This seems very likely to be a bug in Chromium. This can't be reproduced in Firefox.
The font-awesome calendar character seems to be causing problems with text-offsets, navigating character by character (right arrow, focus mode) there seems to be a movement timeout occurring at the symbol.

**Comment by Adriani90** [comment]
Yeah this is related to concenteditable atribute, see also #14731. If that one is fixed in Chromium this one will very probably benefit from that as well.

**Comment by Adriani90** [comment]
This is still reproducible in Chrome but also in Firefox:

In Firefox:
1. Pressing down arrow multiple times navigates to the end of the last line "line after", from there pressing up and down arrow skips the line in the middle. That means NVDA navigates only to "line before" or "line after".
```
Right arrow: "blank"
Up arrow: "line before"
Down arrow: "line after"
```

2. Placing the caret at the beginning of the first line "line before", then pressing down arrow, all three lines are navigated. But on the line in the middle NVDA reports only the left side: "today is ..."
```
Left arrow: "L"
Up arrow: "Line before"
Down arrow: "Today is"
Down arrow: "Line after"
```

3. Pressing right arrow on the line in the middle until you reach the date "9/21/2021", then pressing up and down arrow, NVDA navigates all three lines but it reports only the right side of the line in the middle middle "9/21/2021"
```
Right arrow: "9"
Up arrow: "Line before"
Down arrow: "9/21/2021"
Down arrow: "Line after"
```

Expected behavior no matter where the caret is positioned on a line:
```
Up arrow: "Line before"
Down Arrow: "Today is 9/21/2021"
Down arrow: "Line after"
Down arrow: "Line after"
Up arrow: "today is 9/21/2021"
Up arrow: "Line before"
```

In Chrome, the line in the middle is always read completely as expected: "Today is  9/21/2021", 
but pressing up and down arrow key moves the caret in a confusing way. There is no way to navigate all three lines just by pressing up and down arrow key.


cc: @jcsteh, @SaschaCowley, @aleventhal I guess this needs to be fixed consistently by browser developres. Right?

**Comment by Adriani90** [comment]
By the way aria-hidden doesn't seem to have any effect on how the caret is moved, in my case both text boxes behave the same.
